### PR TITLE
decouple amb from observable

### PR DESCRIPTION
**Pull request by gchudnov**
Decoupled `amb` from observable, please review.

Thank you,
Grigoriy

### CODE REVIEW

File: Rx/v2/src/rxcpp/operators/rx-amb.hpp [hunk 1]
```diff
@@ -172,35 +216,88 @@ struct amb
...
+template<class... AN>
+auto amb(AN&&... an)
+    ->     operator_factory<amb_tag, AN...> {
+    return operator_factory<amb_tag, AN...>(std::make_tuple(std::forward<AN>(an)...));
+}
+
+}
+
+template<class SourceObservable, class Observable, class Coordination>
+struct defer_amb : public defer_observable<
```
Comment (kirkshoop) [inline Rx/v2/src/rxcpp/operators/rx-amb.hpp]: are these defer constructs still required? I was hoping that the new pattern would not require these.

Comment (gchudnov) [inline Rx/v2/src/rxcpp/operators/rx-amb.hpp]: Yes, they should be redundant.
Let me remove them.

Comment (gchudnov) [inline Rx/v2/src/rxcpp/operators/rx-amb.hpp]: Removed, please take a look.
...and I hope it compiles on msvc2013 :)

Comment (kirkshoop) [inline Rx/v2/src/rxcpp/operators/rx-amb.hpp]: It worked!

**Comment by kirkshoop** [comment]
Thank you!

**Pull request merged**